libraries/AP_HAL_ChibiOS/AnalogIn.cpp

```diff
@@ -289,6 +293,119 @@ void AnalogIn::read_adc(uint32_t *val)
     chSysUnlock();
 }
 
+
+#if HAL_WITH_MCU_MONITORING
+/*
+  on H7 we can support monitoring MCU temperature and voltage using ADC3
+ */
+#define ADC3_GRP1_NUM_CHANNELS 3
+
+// internal ADC channels (from H7 reference manual)
+#define ADC3_VSENSE_CHAN 18
+#define ADC3_VREFINT_CHAN 19
+#define ADC3_VBAT4_CHAN 17
+
+// samples filled in by ADC DMA engine
+adcsample_t *AnalogIn::samples_adc3;
+uint32_t AnalogIn::sample_adc3_sum[ADC3_GRP1_NUM_CHANNELS];
+// we also keep min and max so we can report the range of voltages
+// seen, to give an idea of supply stability
+uint16_t AnalogIn::sample_adc3_max[ADC3_GRP1_NUM_CHANNELS];
+uint16_t AnalogIn::sample_adc3_min[ADC3_GRP1_NUM_CHANNELS];
+uint32_t AnalogIn::sample_adc3_count;
+
+/*
+  callback from ADC3 driver when sample buffer is filled
+ */
+void AnalogIn::adc3callback(ADCDriver *adcp)
+{
+    const adcsample_t *buffer = samples_adc3;
```

Comment:
should check and return if null

Reply:
as it now skips the init if alloc fails then the callback is never setup